keyrings/codeartifact.py

```diff
@@ -193,3 +187,28 @@ def set_password(self, service, username, password):
     def delete_password(self, service, username):
         # Defer deleting a password to the next backend
         raise NotImplementedError()
+
+    @staticmethod
+    def get_boto_session(*, region, config):
+        # Create session with any supplied configuration.
+        session = boto3.Session(
+            region_name=region,
+            profile_name=config.get("profile_name"),
+            aws_access_key_id=config.get("aws_access_key_id"),
+            aws_secret_access_key=config.get("aws_secret_access_key"),
+        )
+
+        if "assume_role" in config:
+            sts_client = session.client("sts")
+            assumed_role_object = sts_client.assume_role(
+                RoleArn=config["assume_role"],
+                RoleSessionName="KeyRingsCodeArtifact",
```

Comment:
The `RoleSessionName` shouldn't be hard-coded. The AWS credentials file uses `role_session_name` so what about `assume_role_session_name`? If it's not explicitly configured (ie: with a default of `None`) then `boto3` should generate one by default.